lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -496,13 +433,9 @@ def collect_mixed_in_class_methods(constant, dynamic_extends)
           Array(mixed_in_module)
         end
 
-        sig { params(tree: RBI::Tree, constant: Module).void }
-        def compile_mixes_in_class_methods(tree, constant)
-          return if constant.is_a?(Class)
-
-          dynamic_extends, dynamic_includes = collect_dynamic_mixins_of(constant)
-
-          dynamic_includes
+        sig { params(tree: RBI::Tree, constant: Module, mixin_analyzer: DynamicMixinCompiler).void }
+        def compile_mixes_in_class_methods(tree, constant, mixin_analyzer)
```

Comment:
Now that I think about it, I think we can move this method to `DynamicMixinCompiler` as well.

We need to:

1. Get rid of old Sorbet support and bump our min Sorbet dependency in the gemspec to `> 0.5.6200` https://github.com/Shopify/tapioca/blob/main/lib/tapioca/compilers/sorbet.rb/#L20-L22
2. That means we don't need the whole `collect_mixed_in_class_methods` method, we can just grab the code in here: https://github.com/Shopify/tapioca/blob/main/lib/tapioca/compilers/symbol_table/symbol_generator.rb/#L471-L473 and inline it into this function.
3. After calling this method on `mixin_compiler`, we can  then use the `dynamic_includes` and `dynamic_extends` attributes of `mixin_compiler` and add their symbols to the queue.

So I propose that we change `compile_body` to do:
```ruby
def compile_body(tree, name, constant)
  # ...
  compile_props(tree, constant)
  compile_enums(tree, constant)
  compile_dynamic_mixins(tree, constant)
end
```
and then, we can have:
```ruby
def compile_dynamic_mixins(tree, constant)
  # this is only valid for pure modules
  return if constant.is_a?(Class)

  mixin_compiler = DynamicMixinCompiler.new(constant)
  # we don't need to pass `constant` since the mixin compiler already has it, it can store it.
  mixin_compiler.compile_mixes_in_class_methods(tree) 
  mixin_compiler.compile_class_attributes(tree)

  (mixin_compiler.dynamic_includes + mixin_compiler.dynamic_extends).each do |mod|
    add_to_symbol_queue(name_of(mod))
  end
end
```

Reply:
I really like this refactor and it's indeed easier than we initially thought to move all of this inside the new mixin compiler. However, I did move the entire `collect_mixed_in_class_methods` inside as well. Changing the Sorbet dependency requirement to `0.5.6200` is a pretty big bump considering that we are currently on `0.4.4471`.

What do you think?